query/src/main/java/org/infinispan/search/mapper/search/loading/context/impl/InfinispanLoadingContext.java

```diff
@@ -88,7 +82,8 @@ public int hashCode() {
       }
 
       @Override
-      public PojoSelectionEntityLoader<E> createLoader(Set<? extends PojoLoadingTypeContext<? extends E>> expectedTypes) {
+      public PojoSelectionEntityLoader<E> createEntityLoader(Set<? extends PojoLoadingTypeContext<? extends E>> set,
+                                                             PojoSelectionLoadingContext pojoSelectionLoadingContext) {
          return entityLoader;
```

Comment:
Do something like this instead:

```suggestion
                                                             PojoSelectionLoadingContext context) {
         return ((InfinispanLoadingContext) context).entityLoader;
```

(yes the visibility of `entityLoader` will have to change)